awx/api/serializers.py

```diff
@@ -2981,104 +3018,160 @@ class Meta:
+        if field_name == 'credential':
+            return self.build_standard_field(field_name,
+                                             self.credential)
+        return super(WorkflowJobTemplateNodeSerializer, self).build_field(field_name, info, model_class, nested_depth)
 
     def validate(self, attrs):
-        if 'char_prompts' in attrs:
-            if 'job_type' in attrs['char_prompts']:
-                job_types = [t for t, v in JOB_TYPE_CHOICES]
-                if attrs['char_prompts']['job_type'] not in job_types:
-                    raise serializers.ValidationError({
-                        "job_type": _("%(job_type)s is not a valid job type. The choices are %(choices)s.") % {
-                            'job_type': attrs['char_prompts']['job_type'], 'choices': job_types}})
+        deprecated_fields = {}
+        if 'credential' in attrs:
+            deprecated_fields['credential'] = attrs.pop('credential')
+        view = self.context.get('view')
         if self.instance is None and ('workflow_job_template' not in attrs or
                                       attrs['workflow_job_template'] is None):
             raise serializers.ValidationError({
                 "workflow_job_template": _("Workflow job template is missing during creation.")
             })
-        ujt_obj = attrs.get('unified_job_template', None)
+        if 'unified_job_template' in attrs:
+            ujt_obj = attrs['unified_job_template']
+        elif self.instance:
+            ujt_obj = self.instance.unified_job_template
+        else:
+            raise serializers.ValidationError({
+                "unified_job_template": _("Node needs to have a template attached.")})
         if isinstance(ujt_obj, (WorkflowJobTemplate, SystemJobTemplate)):
             raise serializers.ValidationError({
                 "unified_job_template": _("Cannot nest a %s inside a WorkflowJobTemplate") % ujt_obj.__class__.__name__})
-        return super(WorkflowJobTemplateNodeSerializer, self).validate(attrs)
+        attrs = super(WorkflowJobTemplateNodeSerializer, self).validate(attrs)
+        accepted, rejected, errors = ujt_obj._accept_or_ignore_job_kwargs(**self._build_mock_obj(attrs).prompts_dict())
+        # Do not raise survey validation errors
+        errors.pop('variables_needed_to_start', None)
+        if errors:
+            raise serializers.ValidationError(errors)
+        if 'credential' in deprecated_fields:
+            cred = deprecated_fields['credential']
+            attrs['credential'] = cred
+            if cred is not None:
+                cred = Credential.objects.get(pk=cred)
+                view = self.context.get('view', None)
+                if (not view) or (not view.request) or (view.request.user not in cred.use_role):
+                    raise PermissionDenied()
+        return attrs
+
+    def create(self, validated_data):
+        deprecated_fields = {}
+        if 'credential' in validated_data:
```

Comment:
May want TODOs on some of this backwards-combat stuff signaling that we can drop it at some point when we remove support for this.